waiter/src/waiter/async_request.clj

```diff
@@ -65,10 +65,17 @@
                 (log/info "request has been cleared from store, exiting monitoring loop")
                 (complete-async-request :success)
                 (if (= trigger-chan exit-chan) :request-terminated :request-no-longer-active))
-              (let [{:keys [body headers error status]} (async/<! (make-http-request))]
+              (let [reservation-status-promise (promise)
+                    {:keys [body headers error status]} (async/<! (make-http-request reservation-status-promise))]
                 (when body
                   (async/close! body))
-                (if error
+                (cond
+                  (realized? reservation-status-promise)
```

Comment:
Do we need to introduce a new way to pass errors in addition to returning exceptions as before?